storage/rocksdb/rdb_datadic.cc

```diff
@@ -3137,6 +3147,29 @@ bool Rdb_tbl_def::put_dict(Rdb_dict_manager *const dict,
   return false;
 }
 
+// Length that each index flag takes inside the record.
+// Each index in the array maps to the enum INDEX_FLAG
+static const std::array<int, 1> index_flag_lengths = {
+    {ROCKSDB_SIZEOF_TTL_RECORD}};
+
+uint32 Rdb_key_def::calculate_index_flag_offset(uint32 index_flags,
+                                                enum INDEX_FLAG flag) {
+  uint offset = 0;
+  for (size_t bit = 0; bit < sizeof(index_flags); ++bit) {
+    int mask = 1 << bit;
+    if (index_flags & mask) {
+      /* Exit once we've reached the proper flag */
+      if (flag == mask) {
```

Comment:
This check should be outside the index_flags & mask conditional. It should probably be an error perhaps if the index_flag bit isn't set for the flag we're looking for, but this function is calculating the offset of where the data would have started from. I could see us using this function to skip over all of the metatags to get to the data. In which case, maybe INDEX_FLAG should have a max enum. If that is requested, then we return whatever the offset is at that point in time.